libgnucash/engine/test/gtest-gnc-datetime.cpp

```diff
@@ -130,6 +130,45 @@ TEST(gnc_date_constructors, test_str_format_constructor)
         {   "m-d",        "6'8",   curr_year,  6,  8},
         {   "m-d",       "0801",   curr_year,  8,  1},
 
+        { "d-m-y", "1 Aug 2013",        2013,  8,  1},
+        { "d-m-y", "1 Sep 2013",        2013,  9,  1},
+        { "d-m-y", "1 September 2013",  2013,  9,  1},
+        { "d-m-y", "04/11/2009",        2009, 11,  4},
+        { "d-m-y",  "12.3.1985",        1985,  3, 12},
+
+        { "d-m-y", "0 Aug 2013",          -1, -1, -1},
```

Comment:
A nitpick: I had separated the successful samples from the failing samples, adding comments to the failing ones to explain why they should fail. You have interleaved successful and failing samples. I don't have strong preference over the ordering, but I do like the consistency. So please either move your failing samples to the end of the other failing samples list or interleave my failing samples with my successful ones as well.